index.d.ts

```diff
@@ -0,0 +1,91 @@
+/**
+ * Syntactic units in unist syntax trees are called nodes.
+ */
+export interface Node {
+  /**
+   * the variant of a node.
+   */
+  type!: string;
```

Comment:
Is there some way to say that `type` cannot be the empty string?

Reply:
Not that I am aware of, not null was the closest constraint I could find.
We could add a doc code comment noting not empty.